packages/api-server/src/filter-web3-tx.ts

```diff
@@ -0,0 +1,359 @@
+    };
+
+    return [ethTx, receipt];
+  } else if (
+    toId === CKB_SUDT_ACCOUNT_ID &&
+    toScript.code_hash === envConfig.l2SudtValidatorScriptTypeHash
+  ) {
+    const sudtArgs = new schemas.SUDTArgs(new Reader(l2Tx.raw.args));
+    if (sudtArgs.unionType() === "SUDTTransfer") {
+      const sudtTransfer: schemas.SUDTTransfer = sudtArgs.value();
+      const toAddress = new Reader(sudtTransfer.getTo().raw()).serializeJson();
+      if (toAddress.length !== 42) {
+        return undefined;
+      }
+      const amount = Uint128.fromLittleEndian(
+        new Reader(sudtTransfer.getAmount().raw()).serializeJson()
+      );
+      const fee = Uint128.fromLittleEndian(
+        new Reader(sudtTransfer.getFee().raw()).serializeJson()
+      );
+      const value: Uint128 = amount;
+      const gasPrice: Uint128 = new Uint128(1n);
+      const gasLimit: Uint128 = fee;
+
+      const ethTx: EthTransaction = {
+        blockHash: l2TxReceipt ? pendingBlockHash : null,
+        blockNumber: l2TxReceipt ? pendingBlockNumber : null,
+        transactionIndex: l2TxReceipt ? PENDING_TRANSACTION_INDEX : null,
+        from: fromAddress,
+        gas: gasLimit.toHex(),
+        gasPrice: gasPrice.toHex(),
+        hash: txHash,
+        input: "0x",
+        nonce,
+        to: toAddress,
+        value: value.toHex(),
+        v: v.toHex(),
+        r,
+        s,
+      };
+
+      const receipt: EthTransactionReceipt = {
+        transactionHash: txHash,
+        transactionIndex: PENDING_TRANSACTION_INDEX,
+        blockHash: pendingBlockHash,
+        blockNumber: pendingBlockNumber,
+        from: fromAddress,
+        to: toAddress,
+        gasUsed: gasLimit.toHex(),
+        cumulativeGasUsed: gasLimit.toHex(),
+        logsBloom: DEFAULT_LOGS_BLOOM,
+        logs: [],
+        contractAddress: null,
+        status: "0x1",
+      };
+
+      return [ethTx, receipt];
+    }
+  }
+
+  return undefined;
+}
+
+export interface PolyjuiceArgs {
+  isCreate: boolean;
+  gasLimit: HexNumber;
+  gasPrice: HexNumber;
+  value: HexNumber;
+  inputSize: HexNumber;
+  input: HexString;
+}
+
+function decodePolyjuiceArgs(args: HexString): PolyjuiceArgs {
+  const buf = Buffer.from(args.slice(2), "hex");
+
+  const isCreate = buf[7].toString(16) === "3";
+  const gasLimit = Uint64.fromLittleEndian(
+    "0x" + buf.slice(8, 16).toString("hex")
+  ).toHex();
+  const gasPrice = Uint128.fromLittleEndian(
+    "0x" + buf.slice(16, 32).toString("hex")
+  ).toHex();
+  const value = Uint128.fromLittleEndian(
+    "0x" + buf.slice(32, 48).toString("hex")
+  ).toHex();
+
+  const inputSize = Uint32.fromLittleEndian(
+    "0x" + buf.slice(48, 52).toString("hex")
+  );
+
+  const input = "0x" + buf.slice(52, 52 + inputSize.getValue()).toString("hex");
+
+  return {
+    isCreate,
+    gasLimit,
+    gasPrice,
+    value,
+    inputSize: inputSize.toHex(),
+    input,
+  };
+}
+
+function accountIdToEthAddress(accountScriptHash: Hash, id: U32): HexString {
```

Comment:
do we still use this method? isn't contract address  = short_address = script_hash.slice(0, 42) ?

Reply:
updated.